src/clients/ApplicationClient.js

```diff
@@ -704,7 +704,130 @@ export default class ApplicationClient extends BaseClient {
     this.log.debug("[ApplicationClient] getAllDevices() - BULK");
     return this.callApi('GET', 200, true, ["bulk", "devices"], null, params);
   }
-
+	
+   /**
+   * Gateway Access Control (Beta)
+   * The methods in this section follow the documentation listed under the link:
+   * https://console.ng.bluemix.net/docs/services/IoT/gateways/gateway-access-control.html#gateway-access-control-beta-
+   * Involves the following sections from the above mentioned link:
+   * Assigning a role to a gateway
+   * Adding devices to and removing devices from a resource group
+   * Finding a resource group
+   * Querying a resource group
+   * Creating and deleting a resource group
+   * Updating group properties
+   * Retrieving and updating device properties
+   * 
+   */
+  
+	//getGatewayGroup(gatewayId)
+	//updateDeviceRoles(deviceId, roles[])
+	//getAllDevicesInGropu(groupId)
+	//addDevicesToGroup(groupId, deviceList[])
+	//removeDevicesFromGroup(groupId, deviceList[])
+  
+  getGatewayGroup(gatewayId){
+    this.log.debug("[ApplicationClient] getGatewayGroup("+gatewayId+")");
+    return this.callApi('GET', 200, true, ['authorization', 'devices' , gatewayId], null);
+  }
+  
+  updateDeviceRoles(deviceId, roles){
+    this.log.debug("[ApplicationClient] updateDeviceRoles("+deviceId+","+roles+")");
+    return this.callApi('PUT', 200, false, ['authorization', 'devices', deviceId, 'roles'], roles);
+  }  
+
+  getAllDevicesInGroup(groupId){
+    this.log.debug("[ApplicationClient] getAllDevicesInGropu("+groupId+")");
+    return this.callApi('GET', 200, true, ['bulk', 'devices' , groupId], null);
+  }
+
+  addDevicesToGroup(groupId, deviceList){
+    this.log.debug("[ApplicationClient] addDevicesToGroup("+groupId+","+deviceList+")");
+    return this.callApi('PUT', 200, false, ['bulk', 'devices' , groupId, "add"], deviceList);
+  }
+
+  removeDevicesFromGroup(groupId, deviceList){
+    this.log.debug("[ApplicationClient] removeDevicesFromGroup("+groupId+","+deviceList+")");
+    return this.callApi('PUT', 200, false, ['bulk', 'devices' , groupId, "remove"], deviceList);
+  }
+
+  // https://console.ng.bluemix.net/docs/services/IoT/gateways/gateway-access-control.html
+  
+	// Finding a Resource Group
+		// getGatewayGroups()
+	// Querying a resource group
+		// getUniqueDevicesInGroup(groupId)
+		// getUniqueGatewayGroup(groupId)
+	// Creating and deleting a resource group
+		// createGatewayGroup(groupName)
+		// deleteGatewayGroup(groupId)
+	// Retrieving and updating device properties
+		// getGatewayGroupProperties()
+		// getDeviceRoles(deviceId)
+		// updateGatewayProperties(gatewayId,control_props)
+		// updateDeviceControlProperties(deviceId, withroles)
+
+  // Finding a Resource Group
+  getGatewayGroups(){
```

Comment:
change to `getGroups()`

Also, this API can be passed a set of search parameters... should we expose that here, or in a different method?  If we use a different method, we could call this one `getAllGroups()` and another one `getGroups(searchParams)`